src/route-recognizer.js

```diff
@@ -79,13 +80,13 @@ export class RouteRecognizer {
 
     let handlers = [{ handler: route.handler, names: names }];
 
+    this.routes.set(route.handler, { segments, handlers });
     if (routeName) {
       let routeNames = Array.isArray(routeName) ? routeName : [routeName];
       for (let i = 0; i < routeNames.length; i++) {
-        this.names[routeNames[i]] = {
-          segments: segments,
-          handlers: handlers
-        };
+        if (!(routeNames[i] in this.names)) {
```

Comment:
I forgot to document this change. If two routes had the same name, e.g. `{ route: ['a','b'], name: 'foo' }`, which is processed as two routes, the latter route would override the former in the name index. This means if you would generate a route for 'foo' above, you would get '#/b' as a result. 

This was a regression. I'm not sure where the regression was introduced, however.